lib/internal/process/coverage.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const path = require('path');
+const { mkdirSync, writeFileSync } = require('fs');
+const inspector = require('inspector');
+// TODO(addaleax): add support for coverage to worker threads.
+const hasInspector = process.config.variables.v8_enable_inspector === 1 &&
+  require('internal/worker').isMainThread;
+
+let session;
+
+function writeCoverage() {
+  if (!hasInspector || !session) {
+    return;
+  }
+
+  const filename = `coverage-${process.pid}-${Date.now()}.json`;
+  try {
+    mkdirSync(process.env.NODE_V8_COVERAGE);
```

Comment:
Should this use the recursive option?

Reply:
I'd like to switch to recursive as soon as [this issue](https://github.com/nodejs/node/pull/22302) is resolved, and we know the shape of the API won't be changing.

This is actually a perfect example of why mkdirp is a good addition to the API :)